### PR TITLE
empty activity in dev

### PR DESCRIPTION
**Pull request by EugeneWalkers**
rewriting layout file

added icon

write an adapter for recyclerview

full realized: add, delete and click

### CODE REVIEW

File: app/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -6,13 +6,27 @@
     android:layout_height="match_parent"
     tools:context=".MainActivity">
 
-    <TextView
+
+    <android.support.v7.widget.RecyclerView
+        android:id="@+id/items"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
```
Comment (pasha656) [inline app/src/main/res/layout/activity_main.xml]: You can use `0dp`

File: app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -6,13 +6,27 @@
...
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:layout_editor_absoluteX="8dp"
+        tools:layout_editor_absoluteY="8dp" />
```
Comment (pasha656) [inline app/src/main/res/layout/activity_main.xml]: Thy you need tools ?

File: app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MainActivity.java [hunk 3]
```diff
@@ -1,13 +1,94 @@
...
 public class MainActivity extends AppCompatActivity {
 
+    private final List<String> items = new ArrayList<>();
+    private MyRecyclerAdapter adapter;
+    private AlertDialog dialogToInputData;
+
+
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
```
Comment (pasha656) [inline app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MainActivity.java]: Please, do not forget about `final`

File: app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MainActivity.java [hunk 4]
```diff
@@ -1,13 +1,94 @@
...
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        setItems();
+        setAlertDialog();
+        setAddButton();
```
Comment (pasha656) [inline app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MainActivity.java]: Bad name

File: app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MainActivity.java [hunk 5]
```diff
@@ -1,13 +1,94 @@
...
+        items.add("Two");
+        items.add("Three");
+        items.add("Four");
+        items.add("Five");
+        items.add("Six");
+        items.add("Seven");
+    }
+
+    private void setRecyclerView() {
+        RecyclerView recycler = findViewById(R.id.items);
```
Comment (pasha656) [inline app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MainActivity.java]: `final`

File: app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MainActivity.java [hunk 6]
```diff
@@ -1,13 +1,94 @@
...
+        recycler.setLayoutManager(new LinearLayoutManager(this));
+        adapter = new MyRecyclerAdapter(items);
+        adapter.setOnItemClickListener(new MyRecyclerAdapter.ClickListener() {
+            @Override
+            public void onClick(View view, int pos) {
```
Comment (pasha656) [inline app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MainActivity.java]: final

File: app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MainActivity.java [hunk 7]
```diff
@@ -1,13 +1,94 @@
...
+                Toast.makeText(MainActivity.this, items.get(pos), Toast.LENGTH_SHORT).show();
+            }
+
+            @Override
+            public void onLongClick(View view, int pos) {
+                items.remove(pos);
+                adapter.notifyDataSetChanged();
```
Comment (pasha656) [inline app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MainActivity.java]: Replace to `adapter.notifyItemChanged()`

File: app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MyRecyclerAdapter.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import java.util.List;
+
+public class MyRecyclerAdapter extends RecyclerView.Adapter<MyRecyclerAdapter.MyViewHolder> {
+
+    private List<String> items;
```
Comment (pasha656) [inline app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MyRecyclerAdapter.java]: Please, do not forget about `final` in all cases

File: app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MyRecyclerAdapter.java [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+    @NonNull
+    @Override
+    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
+        MyViewHolder holder = new MyViewHolder(new TextView(viewGroup.getContext()));
+        return holder;
+    }
```
Comment (pasha656) [inline app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MyRecyclerAdapter.java]: I would like suggest you this solution: 

`final View view = new TextView(viewGroup.getContext());
        final RecyclerView.ViewHolder holder = new MyViewHolder(view);
        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(final View view) {
                final int adapterPosition = holder.getAdapterPosition();
                if (adapterPosition != RecyclerView.NO_POSITION) {
                    clicker.onClick(items[adapterPosition]);
                }
            }
        });
        return (MyViewHolder) holder;`

File: app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MyRecyclerAdapter.java [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+public class MyRecyclerAdapter extends RecyclerView.Adapter<MyRecyclerAdapter.MyViewHolder> {
+
+    private List<String> items;
+
+    @NonNull
+    @Override
+    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
+        MyViewHolder holder = new MyViewHolder(new TextView(viewGroup.getContext()));
+        return holder;
```
Comment (pasha656) [inline app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MyRecyclerAdapter.java]: I would like you suggest this solution

final View view = new TextView(viewGroup.getContext());
        final RecyclerView.ViewHolder holder = new MyViewHolder(view);
        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(final View view) {
                final int adapterPosition = holder.getAdapterPosition();
                if (adapterPosition != RecyclerView.NO_POSITION) {
                    clicker.onClick(items[adapterPosition]);
                }
            }
        });
        return (MyViewHolder) holder;

File: app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MyRecyclerAdapter.java [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+import java.util.List;
+
+public class MyRecyclerAdapter extends RecyclerView.Adapter<MyRecyclerAdapter.MyViewHolder> {
+
+    private List<String> items;
+
+    @NonNull
+    @Override
+    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
+        MyViewHolder holder = new MyViewHolder(new TextView(viewGroup.getContext()));
```
Comment (pasha656) [inline app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MyRecyclerAdapter.java]: Please, create separate layout for this

File: app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MyRecyclerAdapter.java [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    static class MyViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener, View.OnLongClickListener {
+
+        TextView textView;
+
+        MyViewHolder(@NonNull View itemView) {
+            super(itemView);
+            textView = ((TextView) itemView);
+            textView.setTextSize(40);
+            textView.setOnClickListener(this);
+            textView.setOnLongClickListener(this);
```
Comment (pasha656) [inline app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MyRecyclerAdapter.java]: Please, create separate layout and set this attr here

File: app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MyRecyclerAdapter.java [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+    interface ClickListener {
+        void onClick(View view, int pos);
+        void onLongClick(View view, int pos);
+    }
+
+    public void setOnItemClickListener(ClickListener listener) {
+        clicker = listener;
+    }
+
+    private static ClickListener clicker;
```
Comment (pasha656) [inline app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MyRecyclerAdapter.java]: Why static ?

File: app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MainActivity.java [hunk 14]
```diff
@@ -1,13 +1,94 @@
...
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import java.util.ArrayList;
+import java.util.List;
 
 public class MainActivity extends AppCompatActivity {
```
Comment (pasha656) [inline app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MainActivity.java]: Better `implements MyRecyclerAdapter.ClickListener`

and override all method

File: app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MyRecyclerAdapter.java [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+
+    static class MyViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener, View.OnLongClickListener {
+
+        TextView textView;
+
+        MyViewHolder(@NonNull View itemView) {
+            super(itemView);
+            textView = ((TextView) itemView);
+            textView.setTextSize(40);
+            textView.setOnClickListener(this);
```
Comment (pasha656) [inline app/src/main/java/com/example/yauheni_skarakhodau/epamlearnwork/MyRecyclerAdapter.java]: Not only `textView` should clicked. All container must handle click

File: app/src/main/res/layout/activity_main.xml (hunk 2 above)
Comment (EugeneWalkers) [reply app/src/main/res/layout/activity_main.xml]: done

**Pull request merged**